dimensiondata/src/test/java/org/jclouds/dimensiondata/cloudcontrol/features/ServerApiMockTest.java

```diff
@@ -0,0 +1,230 @@
+                  "administratorPassword", Lists.<Disk>newArrayList(), createServerOptions);
+      assertEquals(serverId, "7b62aae5-bdbe-4595-b58d-c78f95db2a7f");
+      RecordedRequest recordedRequest = assertSent(POST,
+            "/caas/2.4/6ac1e746-b1ea-4da5-a24e-caf1a978789d/server/deployServer");
+      assertBodyContains(recordedRequest, "\"cpu\":{\"count\":1,\"speed\":\"HIGHPERFORMANCE\",\"coresPerSocket\":2}");
+   }
+
+   public void testClone() throws Exception {
+      server.enqueue(jsonResponse("/cloneServerResponse.json"));
+      CloneServerOptions cloneServerOptions = CloneServerOptions.builder().description("description")
+            .clusterId("EU6-02").guestOsCustomization(true).build();
+      final String imageId = serverApi()
+            .cloneServer("9ed47330-5561-11e5-8c14-b8ca3a5d9ef8", "serverNewImageName", cloneServerOptions);
+      assertEquals(imageId, "3389ffe8-c3fc-11e3-b29c-001517c4643e");
+
+      RecordedRequest recordedRequest = assertSent(POST,
+            "/caas/2.4/6ac1e746-b1ea-4da5-a24e-caf1a978789d/server/cloneServer");
+
+      assertBodyContains(recordedRequest,
+            "{\"id\":\"9ed47330-5561-11e5-8c14-b8ca3a5d9ef8\",\"imageName\":\"serverNewImageName\","
+                  + "\"description\":\"description\",\"clusterId\":\"EU6-02\",\"guestOsCustomization\":true}");
+   }
+
+   public void testGetServerReturnsResourceNotFound() throws InterruptedException {
+      server.enqueue(responseResourceNotFound());
+      Server found = serverApi().getServer("12345");
+      assertNull(found);
+      assertEquals(server.getRequestCount(), 2);
+      assertSent(GET, "/caas/2.4/6ac1e746-b1ea-4da5-a24e-caf1a978789d/server/server/12345");
+   }
+
+   public void testListServers() throws Exception {
+      server.enqueue(jsonResponse("/servers.json"));
+      List<Server> servers = serverApi().listServers().concat().toList();
+      Uris.UriBuilder uriBuilder = Uris.uriBuilder("/caas/2.4/6ac1e746-b1ea-4da5-a24e-caf1a978789d/server/server");
+      Set<String> zones = ctx.utils().injector().getInstance(ZoneIdsSupplier.class).get();
+      for (String zone : zones) {
+         uriBuilder.addQuery("datacenterId", zone);
+      }
+      assertSent(GET, uriBuilder.toString());
+      assertEquals(servers.size(), 1);
+      for (Server s : servers) {
+         assertNotNull(s);
+      }
+   }
+
+   public void testGetServer() throws Exception {
+      server.enqueue(jsonResponse("/server.json"));
+      Server found = serverApi().getServer("12345");
+      assertSent(GET, "/caas/2.4/6ac1e746-b1ea-4da5-a24e-caf1a978789d/server/server/12345");
+      assertNotNull(found);
+      assertNotNull(found.guest().vmTools());
+   }
+
+   public void testGetServer_404() throws Exception {
+      server.enqueue(response404());
+      serverApi().getServer("12345");
+      assertSent(GET, "/caas/2.4/6ac1e746-b1ea-4da5-a24e-caf1a978789d/server/server/12345");
+   }
```

Comment:
Duplicates the `testGetServerReturnsResourceNotFound` test?

Reply:
Yes it looks like it is a duplicate. I will remove testGetServerReturnsResourceNotFound